src/Resolution.ts

```diff
@@ -78,6 +79,27 @@ export default class Resolution {
     }
   }
 
+  /**
+   * Creates a resolution with configured infura id for ens and cns
+   * @param infura infura project id
+   */
+  static infura(infura: string): Resolution {
+    return new this({
+      blockchain: {
+        ens: { url: signedInfuraLink(infura), network: 'mainnet' },
+        cns: { url: signedInfuraLink(infura), network: 'mainnet' }
+      }
+    });
+  }
+
+  /**
+   * Creates a resolution instance with configured provider
+   * @param provider - any provider with sendAsync function impelmented
+   */
+  static provider(provider: Web3Provider): Resolution {
+    return new this({blockchain: {web3Provider: provider}});
```

Comment:
cc @mvlabat . I remember something like that didn't work on the website. Any ideas why it is working here?

Reply:
What didn't work exactly? `new this` in a static method? I can't see why this code shouldn't work here.. It's not an abstract class or anything